### PR TITLE
De-clutter variables table (prefill columns)

### LINKED ISSUE #4555: Merge prefill columns into single column on variables tab
Plugin / attribute / role now take up too much space by each having their own column. Instead, make a single column with edit link that opens modal for these fields.

### PR DESCRIPTION
**Pull request by sergei-maertens**
Closes #4555

**Changes**

* Added some structure to the icon components
* Moved prefill columns/form elements into a preview / form modal
* Updated some language/translations to make things less confusing

**Checklist**

Check off the items that are completed or not relevant.

- Impact on features

  - [x] Checked copying a form
  - [x] Checked import/export of a form
  - [x] Config checks in the configuration overview admin page
  - [x] Problem detection in the admin email digest is handled

- Release management

  - [x] I have labelled the PR as "needs-backport" accordingly

- I have updated the translations assets (you do NOT need to provide translations)

  - [x] Ran `./bin/makemessages_js.sh`
  - [x] Ran `./bin/compilemessages_js.sh`

- Commit hygiene

  - [x] Commit messages refer to the relevant Github issue
  - [x] Commit messages explain the "why" of change, not the how

### CODE REVIEW

**Comment by sergei-maertens** [comment]
@stevenbal I recommend going commit-by-commit for the review!

File: src/openforms/js/components/admin/form_design/variables/VariablesEditor.stories.js [hunk 1]
```diff
@@ -507,7 +507,7 @@ export const ConfigurePrefill = {
     await userEvent.click(userDefinedVarsTab);
 
     // open modal for configuration
-    const editIcon = canvas.getByTitle('Edit prefill configuration');
+    const editIcon = canvas.getByTitle('Prefill instellen');
```
Comment (stevenbal) [inline src/openforms/js/components/admin/form_design/variables/VariablesEditor.stories.js:510]: I thought it was customary to use English and then add the Dutch translation (at least for Python code)? Is this different for JS?

File: src/openforms/js/lang/en.json [hunk 2]
```diff
@@ -399,6 +394,11 @@
     "description": "Logic type selection label",
     "originalDefault": "Please select the logic type:"
   },
+  "9LdZVP": {
+    "defaultMessage": "Attribute",
+    "description": "Variable prefill attribute label",
+    "originalDefault": "Attribute"
+  },
```
Comment (stevenbal) [inline src/openforms/js/lang/en.json:401]: I assume the actual translations will be added in a separate PR?

File: src/openforms/js/components/admin/form_design/variables/VariablesTable.js [hunk 3]
```diff
@@ -2,19 +2,17 @@ import classNames from 'classnames';
 import PropTypes from 'prop-types';
```
[the remark this answers is not in the document]
Comment (sergei-maertens) [reply src/openforms/js/components/admin/form_design/variables/VariablesTable.js:2]: yeah - for now you still have to edit this on the component itself (in the prefill tab), but for user-defined variables you can edit it in the table here already

We will move the configuration from the formio-builder (e.g. https://open-formulieren.github.io/formio-builder/?path=/story/public-api-componentconfiguration--default -> Prefill tab) to the variables table in due time.

Actually a good reminder to update the translations there to be consistent.

File: src/openforms/js/components/admin/form_design/variables/prefill/PrefillSummary.js [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+            description: "'Edit variable prefill' icon label",
+          })}
+          onClick={() => setModalOpen(!modalOpen)}
+        />
+      )}
+    </div>
+  );
+
+  return (
+    <div style={{display: 'flex', justifyContent: 'space-between', alignItems: 'center'}}>
```
[the remark this answers is not in the document]
Comment (sergei-maertens) [reply src/openforms/js/components/admin/form_design/variables/prefill/PrefillSummary.js:59]: yes, though there are pre-existing situations where the same thing is applied, so I'd prefer to do this in a separate PR

File: src/openforms/js/components/admin/form_design/variables/prefill/PrefillSummary.js [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+      {plugin === '' && attribute === '' ? (
+        '-'
+      ) : (
+        <FormattedMessage
+          tagName="span"
+          description="Prefill configuration summary"
+          defaultMessage="<code>{attr}</code> from <code>{plugin}</code> (of the {role})"
```
[the remark this answers is not in the document]
Comment (sergei-maertens) [reply src/openforms/js/components/admin/form_design/variables/prefill/PrefillSummary.js:66]: indeed - in the preview we don't have the nice labels available for the values. The old situation also showed the raw value in its own column, so it's not making the situation _worse_.

On the other hand, Joeri was inclined to keep using the raw values because of familiarity with the attributes that already exists, so for now it's a wontfix, maybe even a wonteverfix.

File: src/openforms/js/components/admin/form_design/variables/prefill/PrefillConfigurationForm.js [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+  // reset the attribute value whenever the plugin changes
+  useUpdateEffect(() => {
+    setFieldValue('attribute', '');
+  }, [setFieldValue, value]);
+
+  const choices = availablePrefillPlugins.map(plugin => [plugin.id, plugin.label]);
+  return <Select allowBlank choices={choices} id="id_plugin" {...fieldProps} />;
+};
+
+const AttributeField = () => {
```
[the remark this answers is not in the document]
Comment (sergei-maertens) [reply src/openforms/js/components/admin/form_design/variables/prefill/PrefillConfigurationForm.js:131]: I think the backend validation is a bit shoddy on this right now. Checked with the master branch, and it has the same problems, except it's even a bit worse because it creates the illusion the first attribute is selected while that's not the actual case!

Leaving it blank at least allows us to show no attribute was selected.

I'll make sure to include this in follow up ticket(s).

File: src/openforms/js/components/admin/form_design/variables/prefill/PrefillConfigurationForm.js [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+      allowBlank
+      choices={choices}
+      id="id_attribute"
+      disabled={loading || !plugin}
+      {...fieldProps}
+    />
+  );
+};
+
+const IdentifierRoleField = () => {
```
[the remark this answers is not in the document]
Comment (sergei-maertens) [reply src/openforms/js/components/admin/form_design/variables/prefill/PrefillConfigurationForm.js:167]: the problem is that there is actually no value set in the state at all, and that's why the `allowBlank` is needed. This role can indeed be empty, especially with legacy form exports created before this field was added. The backend code also falls back to `main`, but I'll take a look if I can improve this.

File: src/openforms/js/components/admin/form_design/variables/VariablesEditor.stories.js (hunk 1 above)
Comment (sergei-maertens) [reply src/openforms/js/components/admin/form_design/variables/VariablesEditor.stories.js:510]: same principle applies, however dealing with non-default languages in storybook is a massive pain in the rear, so for these interaction tests we just use the NL strings.

File: src/openforms/js/lang/en.json (hunk 2 above)
Comment (sergei-maertens) [reply src/openforms/js/lang/en.json:401]: English translations are pretty much untouched - we try to get the source strings/defaults right the first time :grimacing:

File: src/openforms/js/components/admin/form_design/variables/prefill/PrefillConfigurationForm.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (stevenbal) [reply src/openforms/js/components/admin/form_design/variables/prefill/PrefillConfigurationForm.js:131]: Check, doing it in a separate ticket is fine by me :+1:

[the remark this answers is not in the document]
Comment (sergei-maertens) [reply src/openforms/js/components/admin/form_design/variables/prefill/PrefillConfigurationForm.js:131]: Created #4563

[the remark this answers is not in the document]
Comment (sergei-maertens) [reply src/openforms/js/components/admin/form_design/variables/prefill/PrefillConfigurationForm.js:131]: oh this particular case actually leads to a server error, interesting. Might fix it in this PR after all.

File: src/openforms/js/components/admin/form_design/variables/prefill/PrefillConfigurationForm.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (sergei-maertens) [reply src/openforms/js/components/admin/form_design/variables/prefill/PrefillConfigurationForm.js:167]: the validation error styling was being mangled by bootstrap, again... So patching this up, and in this case the `allowBlank` can actually be removed since we normalize the formik `initialValues` to address potential missing information.

File: src/openforms/js/components/admin/form_design/variables/prefill/PrefillConfigurationForm.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (sergei-maertens) [reply src/openforms/js/components/admin/form_design/variables/prefill/PrefillConfigurationForm.js:131]: working on this in a separate branch, so I'm marking this as resolved so that we can merge this PR

File: src/openforms/js/lang/en.json (hunk 2 above)
Comment (stevenbal) [reply src/openforms/js/lang/en.json:401]: ah I didn't see this was `en.json`, makes sense then :p
